### PR TITLE
Move CIB into the _cts module

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

File: python/pacemaker/_cts/cib.py [hunk 1]
```diff
@@ -382,43 +379,15 @@ def log(self, args):
...
-        _args.extend(args)
-        setattr(self, methodName, ConfigFactoryItem(*_args, **kargs))
-
-    def unregister(self, methodName):
-        """unregister a constructor"""
-        delattr(self, methodName)
-
-    def createConfig(self, name="pacemaker-1.0"):
-        if name == "pacemaker-1.0":
-            name = "pacemaker10"
```
Comment (kgaillot) [inline python/pacemaker/_cts/cib.py]: I don't see any problem with dropping 1.0, but why?

File: python/pacemaker/_cts/cib.py [hunk 2]
```diff
@@ -34,16 +36,6 @@ def __init__(self, cm, factory, tmpfile=None):
...
-
-class CIB12(ConfigBase):
-    version = "pacemaker-1.2"
-
-    def __init__(self, cm, factory, tmpfile=None):
-        ConfigBase.__init__(self, cm, factory, tmpfile=tmpfile)
-
-        self._counter = 1
-        self._num_nodes = 0
-
```
Comment (kgaillot) [inline python/pacemaker/_cts/cib.py]: Yeah, I don't see any point in having multiple classes, not to mention they unnecessarily restrict us to certain schema versions. We can just take the schema name as an argument. If there's ever anything schema-specific we need to do, we can just parse the version number and compare it as needed.

**Review by kgaillot — COMMENTED** [review]
nice

File: python/pacemaker/_cts/cib.py (hunk 1 above)
Comment (clumens) [reply python/pacemaker/_cts/cib.py]: If you look at [a71f29c](https://github.com/ClusterLabs/pacemaker/pull/3150/commits/a71f29cc28eeeebe202b3d04f660736a1b563372), you'll see there's not actually support for 1.0.  It registers attributes for pacemaker12, pacemaker20, and pacemaker30.  Even though the default is pacemaker-1.0 which maps to the pacemaker10 attribute, there's nothign registered for that.  It'll just end up defaulting to 3.0 anyway.

I could fix that, though.

Comment (kgaillot) [reply python/pacemaker/_cts/cib.py]: Let's just drop the whole subclass thing and have one class that takes the schema name

**Comment by clumens** [comment]
Updated to get rid of the multiple CIB classes and also merge everything from ConfigBase into the new CIB class.  Also pay special attention to the two most recent patches.

**Review by kgaillot — APPROVED** [review]
awesome :)

**Pull request merged**